### PR TITLE
UCP/WIREUP: Fix data race - v1.6

### LINKED ISSUE #3642: shm_ib/test_ucp_tag_mt.send_recv/1 fails on boo31
```
15:59:46 [----------] 1 test from shm_ib/test_ucp_tag_mt
15:59:46 [ RUN      ] shm_ib/test_ucp_tag_mt.send_recv/1
16:00:12 [boo31:19595:0:19595] ucp_proxy_ep.c:214  Assertion `proxy_ep->uct_ep != ((void *)0)' failed
16:00:16 ==19595== brk segment overflow in thread #1: can't grow to 0x4800000
16:00:16 ==19595== (see section Limitations in user manual)
16:01:21 
16:01:21 /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/ucs/debug/assert.c: [ ucs_fatal_error_message() ]
16:01:21       ...
16:01:21        33     }
16:01:21        34 
16:01:21        35     ucs_handle_error(message_buf);
16:01:21 ==>    36     abort();
16:01:21        37 }
16:01:21        38 
16:01:21        39 void ucs_fatal_error_format(const char *file, unsigned line,
16:01:21 
16:02:11 ==== backtrace (tid:  19595) ====
16:02:11  0 0x000000000003c018 ucs_fatal_error_message()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/ucs/debug/assert.c:36
16:02:11  1 0x000000000003c11f ucs_fatal_error_format()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/ucs/debug/assert.c:52
16:02:11  2 0x000000000001d08d ucp_proxy_ep_replace()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/ucp/core/ucp_proxy_ep.c:214
16:02:11  3 0x000000000005d7a9 ucp_signaling_ep_am_short()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/ucp/wireup/signaling_ep.c:56
16:02:11  4 0x0000000000052699 uct_ep_am_short()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/uct/api/uct.h:2100
16:02:11  5 0x0000000000052699 ucp_tag_send_inline()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/ucp/tag/tag_send.c:159
16:02:11  6 0x0000000000052699 ucp_tag_send_nb_inner()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/ucp/tag/tag_send.c:193
16:02:11  7 0x0000000000052699 ucp_tag_send_nb()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../src/ucp/tag/tag_send.c:180
16:02:11  8 0x0000000000696211 test_ucp_tag::send_nb()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/ucp/test_ucp_tag.cc:180
16:02:11  9 0x0000000000696fde test_ucp_tag::send_b()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/ucp/test_ucp_tag.cc:213
16:02:11 10 0x000000000066b504 test_ucp_tag_mt_send_recv_Test::test_body()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/ucp/test_ucp_tag_mt.cc:76
16:02:11 11 0x000000000066b9d1 test_ucp_tag_mt_send_recv_Test::test_body()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/ucp/test_ucp_tag_mt.cc:67
16:02:11 12 0x000000000051407a ucs::test_base::run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/test.cc:276
16:02:11 13 0x00000000005140bd ucs::test_base::TestBodyProxy()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/test.cc:302
16:02:11 14 0x00000000005072ed HandleSehExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:3562
16:02:11 15 0x00000000004fd5ec testing::Test::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:3635
16:02:11 16 0x00000000004fd8b8 testing::TestInfo::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:3812
16:02:11 17 0x00000000004fda7e testing::TestCase::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:3930
16:02:11 18 0x00000000004fda7e testing::TestCase::total_test_count()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:3864
16:02:11 19 0x00000000004fda7e ???()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:3929
16:02:11 20 0x00000000004fe063 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:5802
16:02:11 21 0x00000000004fe063 ???()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:5800
16:02:11 22 0x0000000000506e7d HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:3562
16:02:11 23 0x00000000004fcb39 testing::UnitTest::Run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:5416
16:02:11 24 0x00000000004fcb39 ???()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/gtest-all.cc:5416
16:02:11 25 0x00000000005078c5 main()  /scrap/jenkins/workspace/hpc-ucx-pr/label/boo31/worker/1/contrib/../test/gtest/common/main.cc:101
16:02:11 26 0x000000347981ed1d __libc_start_main()  ???:0
16:02:11 27 0x00000000004e3889 _start()  ???:0
16:02:11 =================================
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/label=boo31,worker=1/10429/console

### PR DESCRIPTION
**Pull request by dmitrygladkov**
## What

Moves sending of `UCP_WIREUP_MSG_ACK` from an async thread (e.g. UD's `uct_ud_iface_timer` that progress data) to `ucp_worker_progress` function.

## Why ?

Fixes #3642
Also, it prevents the calling of UCT AM routines from different threads at the same time

## How ?

Register the function that sends `UCP_WIREUP_MSG_ACK` in UCT worker.

back-port of #3654

### CODE REVIEW

**Comment by shamisp** [comment]
👍

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10539/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10548/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
@yosefe could you review pls?

File: NEWS [hunk 1]
```diff
@@ -25,8 +25,10 @@ Bugfixes:
 - RPM Spec file cleanup
 - Fixing compilation issues with most recent clang and gcc compilers
 - Fixing the wrong name of aliases
+- Fix data race in UCP wireup
 - Fix segfault when libuct.so is reloaded - issue #3558
 
+
```
Comment (yosefe) [inline NEWS]: redundant space line

Comment (dmitrygladkov) [reply NEWS]: @yosefe GH merge conflict solver :( fixed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10554/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
> Test FAILed.
> See [http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10554](http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10554/) for details (Mellanox internal link).

related to #3679

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10568/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10586/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10596/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
@yosefe could you review pls?

**Pull request merged**